File: core/MaximumDifferenceBetweenIncreasingElements/MaximumDifferenceBetweenIncreasingElements.py
from typing import List


def maximumDifference(nums: List[int]) -> int:
    min_num, div_num = nums[0], -1
    for i in range(1, len(nums)):
        if nums[i] > min_num and nums[i] - min_num > div_num:
            div_num = nums[i] - min_num
        min_num = min_num if min_num < nums[i] else nums[i]

    return div_num

File: core/MaximumDifferenceBetweenIncreasingElements/MaximumDifferenceBetweenIncreasingElements.java
package com.dbc;

public class MaximumDifferenceBetweenIncreasingElements {
    public int maximumDifference(int[] nums) {
        int minNum = nums[0], divNum = -1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > minNum && nums[i] - minNum > divNum) {
                divNum = nums[i] - minNum;
            }
            minNum = Math.min(minNum, nums[i]);
        }
        return divNum;
    }
}
